platform/kube/controller.go

```diff
@@ -37,12 +37,15 @@ const (
 	NodeRegionLabel = "failure-domain.beta.kubernetes.io/region"
 	// NodeZoneLabel is the well-known label for kubernetes node zone
 	NodeZoneLabel = "failure-domain.beta.kubernetes.io/zone"
+	// IstioNamespace used by default for Istio cluster-wide installation
+	IstioNamespace = "istio-system"
 )
 
 // ControllerOptions stores the configurable attributes of a Controller.
 type ControllerOptions struct {
 	// Namespace to restrict controller to (empty to disable restriction)
 	Namespace    string
+	AppNamespace string
```

Comment:
please document what that is exactly and how it relates to 'Namespace' and what changed in the meaning of 'Namespace' if anything
I thought (from the code elsewhere) Namespace was IstioNamespace but it seems to be (sometimes?) watched namespace ? for instance what does empty Namespace and non empty AppNamespace do, vs empty,empty and non empty, non empty...

Reply:
Empty "" means all k8s namespaces, it is a k8s constant.

Reply:
k8s.io/apimachinery/pkg/apis/meta/v1.NamespaceAll is ""

Reply:
Again, please explain in the comment what this is and how it is different from `Namespace`